k8s/src/main/scala/io/buoyant/k8s/Api.scala

```diff
@@ -179,6 +180,18 @@ object Watch {
     @JsonIgnore
     def resourceVersion = None
   }
+
+  /**
+   * Extractor allowing `Added` and `Modified` events --- which contain a new
+   * state --- to be handled in one match arm.
+   */
+  object NewState {
```

Comment:
I think extractors are useful when it's extremely obvious from context or naming what they do.  In this case, I don't think it's immediately obvious and I end up jumping around the code to figure out what's going on.  Instead, how about factoring the common match arm bodies into a method and calling that method from both arms?  This avoids duplication while also making it clear that there are two match possibilities that both map to the same behavior.

Reply:
Sure!